task8/TutterApp/src/main/java/dao/PostDAOImpl.java

```diff
@@ -0,0 +1,112 @@
+package dao;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import forms.FilterForm;
+import models.Constants;
+import models.Post;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class PostDAOImpl implements PostDAO {
+    private static List<Post> posts;
+
+    public PostDAOImpl() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.setDateFormat(new SimpleDateFormat(Constants.DATE_FORMAT));
+
+        try {
+            posts = objectMapper.readValue(new File(Constants.PATH), new TypeReference<>() {
+            });
+        } catch (IOException e) {
+            posts = new ArrayList<>();
+
+            System.out.println(e.getMessage());
+        }
+    }
+
+    @Override
+    public Optional<Post> get(Long id) {
+        return posts.stream().filter(post -> post.getId().equals(id)).findAny();
+    }
+
+    @Override
+    public List<Post> getPage(FilterForm filterForm) {
+        return posts.stream().filter(post -> (filterForm.getAuthor() == null || filterForm.getAuthor().equals(post.getAuthor())) &&
```

Comment:
Please always avoid long multi-part comparisons as well as long multiline statements in lambdas. Such as at lines 40 to 43 is plainly unreadable -- can only take your word that it works as it must))

Any statement included in lambda that tends to occypy more than 2 lines of text, must be separated to a private static method and the be called like `Post::correspondsToForm` or `post -> this.corresponds(post, form)`